apps/meteor/client/views/audit/components/SecurityLogsTable.tsx

```diff
@@ -0,0 +1,211 @@
+					</Margins>
+				</Field>
+				<ButtonGroup>
+					<Margins inline={6}>
+						<Button secondary onClick={handleClearFilters}>
+							{t('Clear_filters')}
+						</Button>
+						<Button primary onClick={handleApplyFilters}>
+							{t('Apply_filters')}
+						</Button>
+					</Margins>
+				</ButtonGroup>
+			</Box>
+
+			{isLoading && (
+				<GenericTable>
+					<GenericTableHeader>
+						<GenericTableHeaderCell>{t('Actor')}</GenericTableHeaderCell>
+						<GenericTableHeaderCell>{t('Timestamp')}</GenericTableHeaderCell>
+						<GenericTableHeaderCell>{t('Setting')}</GenericTableHeaderCell>
+						<GenericTableHeaderCell>{t('Changed_from')}</GenericTableHeaderCell>
+						<GenericTableHeaderCell>{t('Changed_to')}</GenericTableHeaderCell>
+					</GenericTableHeader>
+					<GenericTableBody>
+						<GenericTableLoadingRow cols={5} />
+					</GenericTableBody>
+				</GenericTable>
+			)}
+			{isSuccess && data.total === 0 && (
+				<GenericNoResults
+					title={t('No_results_found')}
+					description={t('Try_different_filters')}
+					buttonTitle={t('Clear_filters')}
+					buttonAction={handleClearFilters}
+				/>
+			)}
+			{isSuccess && data.total > 0 && (
+				<GenericTable>
+					<GenericTableHeader>
+						<GenericTableHeaderCell>{t('Actor')}</GenericTableHeaderCell>
+						<GenericTableHeaderCell>{t('Timestamp')}</GenericTableHeaderCell>
+						<GenericTableHeaderCell>{t('Setting')}</GenericTableHeaderCell>
+						<GenericTableHeaderCell>{t('Changed_from')}</GenericTableHeaderCell>
+						<GenericTableHeaderCell>{t('Changed_to')}</GenericTableHeaderCell>
+					</GenericTableHeader>
+					<GenericTableBody>
+						{data.events.map((item) => (
+							<GenericTableRow
+								key={item.ts}
+								role='link'
+								action
+								tabIndex={0}
+								onClick={() =>
+									handleItemClick({
+										actor: item.actor.type === 'user' ? item.actor.username : item.actor.type,
+										timestamp: new Date(item.ts).toDateString(),
+										setting: item.data[0].value,
+										changedFrom: String(item.data[1].value),
+										changedTo: String(item.data[2].value),
+									})
+								}
+							>
+								<GenericTableCell withTruncatedText>
+									<Box display='flex' alignItems='center'>
+										{item.actor.type === 'user' && (
+											<Box mie={4}>
+												<UserAvatar size='x24' username={item.actor.username} />
+											</Box>
+										)}
+										<Box fontScale='p2m' withTruncatedText color='default'>
+											{item.actor.type === 'user' ? item.actor.username : item.actor.type}
+										</Box>
+									</Box>
+								</GenericTableCell>
+								<GenericTableCell withTruncatedText>{item.ts}</GenericTableCell>
+								<GenericTableCell withTruncatedText title={t(String(item.data[0].value))}>
+									{String(item.data[0].value)}
+								</GenericTableCell>
+								<GenericTableCell withTruncatedText>{String(item.data[1].value)}</GenericTableCell>
```

Comment:
You're acessing the data by directly referencing the array position. There's no guarantee that the array will have all the information in the correct order. You need to filter the data by the `key` property.